clientconn.go

```diff
@@ -150,6 +150,8 @@ func WithWriteBufferSize(s int) DialOption {
 
 // WithReadBufferSize lets you set the size of read buffer, this determines how much data can be read at most
 // for each read syscall.
+// A negative value will disable read buffer for a connection so data framer can access the underlying
```

Comment:
Zero seems like a pretty good "disable" value to me..  It seems the problem is that we don't have a "default{Dial,Server}Options", but we do for "defaultCallOptions", right?  Could you look into adding some non-zero defaults that way if it's not too hard?